src/Functions/FunctionsCodingUUID.cpp

```diff
@@ -13,43 +13,159 @@
 #include <Interpreters/Context_fwd.h>
 #include <Interpreters/castColumn.h>
 
-namespace DB
+#include <span>
+
+namespace DB::ErrorCodes
+{
+extern const int ARGUMENT_OUT_OF_BOUND;
+extern const int ILLEGAL_COLUMN;
+extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+extern const int NOT_IMPLEMENTED;
+extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+namespace
+{
+enum class Representation
+{
+    BigEndian,
+    LittleEndian
+};
+
+std::pair<int, int> determineBinaryStartIndexWithIncrement(const ptrdiff_t num_bytes, const Representation representation)
```

Comment:
it is not conventional to pass parameters of value type with const

Reply:
I usually make conscious effort to declare read-only variables `const` in order to improve readability and maintainability. In general, this is in line with the [C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines). My interpretation of the relevant guideline ([Con.1: By default, make objects immutable](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#con1-by-default-make-objects-immutable)) in the context of function parameters is that it's good practice to do so, but it shouldn't be enforced.

What do you think in light of this?

Reply:
I agree, it does make sense, but from my experience people tend to not declare value parameters as `const`, because they are rarely getting changed inside function and doing this makes function declaration longer.
I don't insist on changing it, just a remark

Reply:
It's undeniable that it makes function prototypes longer. In part because of this I never declare value parameters as `const` in function / method declarations, only in actual definitions where it can be reasonably considered a local variable of sorts. Another reason not to use it in declarations is if you decide to revert it to a non-const value the compilation units that include it would have to be recompiled. Anyway, I appreciate you looking into this!